tests/PKSim.Tests/IntegrationTests/FlatParameterInContainsRepositorySpecs.cs

```diff
@@ -68,21 +75,139 @@ public void all_visible_and_editable_parameters_should_be_can_be_varied_except_m
       }
 
       [Observation]
-      public void all_readonly_parameter_should_be_marked_as_non_input()
+      public void all_readonly_parameter_should_be_marked_as_non_input_with_some_known_exceptions()
       {
          var parameters = _allParameters.Where(x => x.ReadOnly)
+            .Where(x => !isOneOfReadOnlyAndIsInputParameters(x.ParameterName))
             .Where(x => x.IsInput).ToList();
 
          parameters.Any().ShouldBeFalse(ErrorMessageFor(parameters));
       }
 
       [Observation]
-      public void all_hidden_parameter_should_be_marked_as_non_input()
+      public void all_hidden_parameter_should_be_marked_as_non_input_with_some_known_exceptions()
       {
          var parameters = _allParameters.Where(x => !x.Visible)
+            .Where(x => !isOneOfReadOnlyAndIsInputParameters(x.ParameterName))
             .Where(x => x.IsInput).ToList();
 
          parameters.Any().ShouldBeFalse(ErrorMessageFor(parameters));
       }
+
+      private void checkIsInputFlag(List<ParameterMetaData> parametersWithWrongIsInputFlag,
+                                    ParameterMetaData parameter, bool isInputShouldBe)
+      {
+         if(parameter.IsInput != isInputShouldBe)
+            parametersWithWrongIsInputFlag.Add(parameter);
+      }
+
+      [Observation]
+      public void should_set_is_input_flag_according_to_specification()
+      {
+         var parametersWithWrongIsInputFlag = new List<ParameterMetaData>();
+
+         foreach (var parameter in _allParameters)
```

Comment:
Could we split that in more tests? This is kind of hard to reason about. Especally since there is a for loop and a continue, at least split by if.

Also usage of `Equals` to compare enum or string is very outdated. use == instead